src/scripts.js

```diff
@@ -113,56 +114,74 @@ function displayFriendGameInfo(id, activityInfo, userStorage, dateString, user)
 }
 
 function postHydrationInputs() {
-  postHydrationData(currentUserId, currentDate, hydrationInput.value)
-    .then((response) => {
-      if (!response.ok) {
-        throw Error(response.statusText);
-      } else {
-        // renderSuccessfulPost("hydration");
-        console.log("success")
-        console.log(currentUserId)
-        hydrationHeader.innerHTML = "Great job! You have submitted your hydration data!"
-      }
-    })
-    .catch(error => {
-      // showPostMessage('fail', error, 'hydration data')
-      console.log(error)
-    })
+  if (hydrationInput.value > 0 && hydrationInput.value < 200) {
+    postHydrationData(currentUserId, currentDate, hydrationInput.value)
+      .then((response) => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        } else {
+          console.log("success")
+          console.log(currentUserId)
+          hydrationHeader.innerText = "Great job! You have submitted your hydration data!"
+          domUpdates.renderSubmittedHydration(hydrationInput.value)
+        }
+      })
+      .catch(error => {
+        hydrationHeader.innerText = "Could not Fetch :( Check Internet?";
+        console.log(error)
+      })
+  } else if (hydrationInput.value > 200) {
+    hydrationHeader.innerText = "Calm down Aquaman - Input too high!"
+  } else {
+    hydrationHeader.innerText = "Please enter a number 0 or higher"
+  }
 }
 
 function postActivityInputs() {
-  postActivityData(currentUserId, currentDate, stepsInput.value, minInput.value, stairsInput.value)
-    .then((response) => {
-      if (!response.ok) {
-        throw Error(response.statusText);
-      } else {
-        // renderSuccessfulPost("hydration");
-        console.log("success")
-        console.log(currentUserId)
-        activityHeader.innerHTML = "Great job! You have submitted your activity data!"
-      }
-    })
-    .catch(error => {
-      // showPostMessage('fail', error, 'hydration data')
-      console.log(error)
-    })
+  if (stepsInput.value > 0 && stepsInput.value < 55000 && minInput.value > 0 && minInput.value < 1440 && stairsInput.value > 0 && stairsInput.value < 2909) {
+    postActivityData(currentUserId, currentDate, stepsInput.value, minInput.value, stairsInput.value)
+      .then((response) => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        } else {
+          console.log("success")
+          console.log(currentUserId)
+          activityHeader.innerText = "Great job! You have submitted your activity data!"
+          domUpdates.renderSubmittedActivity(stepsInput.value, stairsInput.value, minInput.value)
```

Comment:
Nice job invoking the domUpdates in here rather than another .then! I think its cleaner, and likely more reliable.